test/Test.hs

```diff
@@ -0,0 +1,58 @@
+{-# LANGUAGE ExistentialQuantification #-}
+
+module Test(TestSuite(..), Test(..), runTestSuites) where
```

Comment:
Ok, looks well but maybe would be great have a no custom testing framework because could be take issues integrating different tools ( like travis ).

Anyway, at the moment is not a problem and looks great :) Just a comment to take in mind